extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -698,6 +701,13 @@ export class AmpA4A extends AMP.BaseElement {
           checkStillCurrent();
           this.adUrl_ = adUrl;
           this.handleLifecycleStage_('urlBuilt');
+          // If we should skip the XHR, we will instead request and render
+          // by simply writing a frame into the page using
+          // renderViaFrameGet
+          if (this.shouldSkipXhr_()) {
+            this.renderViaFrameGet_(this.adUrl_);
```

Comment:
Instead let's rename this to this.isXhrAllowed() and when false return Promise.reject(XHR_DISALLOWED);  Within catch block you can return null just like NETWORK_FAILURE and should be fine.  You cannot render directly within the promise chain otherwise getPriority will not be honored.  It must be done within layoutCallback.  Instead, can we set the rendering method to direct cache somehow (this.experimentalNonAmpCreativeRenderMethod_)?